src/libsyntax/diagnostics/macros.rs

```diff
@@ -1,112 +1,112 @@
 #[macro_export]
 macro_rules! diagnostic_used {
-    ($code:ident) => (
+    ($code:expr) => (
         let _ = $code;
```

Comment:
@GuillaumeGomez I don't think this macro has any function at all before or after the PR. These constants are all marked as `pub` inside `librustc_error_codes` so they will not be deemed unused if `diagnostic_used!(...);` is missing. Therefore the macro can be removed wholesale and any calls to it. Also, all you've done here is to replace `stringify!($some_ident)` with string literals instead. Keeping the identifiers and continuing to use `stringify!(...)` should work just as well.